rerun_cpp/CMakeLists.txt

```diff
@@ -108,159 +82,33 @@ endif()
-    endif()
-
-    ExternalProject_Add(
-        arrow_cpp
-        PREFIX ${ARROW_DOWNLOAD_PATH}
-        GIT_REPOSITORY https://github.com/apache/arrow.git
-        GIT_TAG apache-arrow-10.0.1
-        GIT_SHALLOW ON
-        GIT_PROGRESS OFF # Git progress sounds like a nice idea but is in practive very spammy.
-
-        # LOG_X ON means that the output of the command will
-        # be logged to a file _instead_ of printed to the console.
-        LOG_CONFIGURE ON
-        LOG_BUILD ON
-        LOG_INSTALL ON
-
-        CMAKE_ARGS
-        --preset ${ARROW_CMAKE_PRESET}
-        -DARROW_BOOST_USE_SHARED=OFF
-        -DARROW_BUILD_SHARED=${ARROW_BUILD_SHARED}
-        -DARROW_BUILD_STATIC=${ARROW_BUILD_STATIC}
-        -DARROW_CXXFLAGS=${ARROW_CXXFLAGS}
-        -DARROW_IPC=ON
-        -DARROW_JEMALLOC=OFF # We encountered some build issues with jemalloc, use mimalloc instead.
-        -DARROW_MIMALLOC=ON
-        -DARROW_USE_ASAN=${ARROW_ASAN}
-        -DARROW_USE_TSAN=OFF
-        -DARROW_USE_UBSAN=OFF
-        -DBOOST_SOURCE=BUNDLED
-        -DCMAKE_INSTALL_PREFIX=${ARROW_DOWNLOAD_PATH}
-        -Dxsimd_SOURCE=BUNDLED
-        -DBOOST_SOURCE=BUNDLED
-        -DARROW_BOOST_USE_SHARED=OFF
-        -DARROW_CXXFLAGS=${DARROW_CXXFLAGS}
-        SOURCE_SUBDIR cpp
-        BUILD_BYPRODUCTS ${ARROW_LIBRARY_FILE} ${ARROW_BUNDLED_DEPENDENCIES_FILE}
-    )
-
-    # arrow_cpp target is not a library. Assemble one from it.
-    if(RERUN_ARROW_LINK_SHARED)
-        add_library(RerunArrowTarget SHARED IMPORTED)
-
-        # For windows we need to know both the dll AND the import library.
-        if(WIN32)
-            set_target_properties(RerunArrowTarget PROPERTIES IMPORTED_IMPLIB ${ARROW_DOWNLOAD_PATH}/lib/arrow.lib)
-        endif()
-    else()
-        add_library(RerunArrowTarget STATIC IMPORTED)
-
-        # Need to set the ARROW_STATIC define, otherwise arrow functions are dllimport decorated on Windows.
-        target_compile_definitions(RerunArrowTarget INTERFACE ARROW_STATIC)
-
-        # We have to explicitly opt in the arrow bundled dependencies, otherwise we're missing the symbols for mimalloc.
-        add_library(RerunArrowTargetBundledDeps STATIC IMPORTED)
-        add_dependencies(RerunArrowTargetBundledDeps arrow_cpp)
-        set_target_properties(RerunArrowTargetBundledDeps PROPERTIES
-            IMPORTED_LOCATION ${ARROW_BUNDLED_DEPENDENCIES_FILE}
-        )
-        target_link_libraries(RerunArrowTarget INTERFACE RerunArrowTargetBundledDeps)
-    endif()
-
-    add_dependencies(RerunArrowTarget arrow_cpp)
-    set_target_properties(RerunArrowTarget PROPERTIES
-        IMPORTED_LOCATION ${ARROW_LIBRARY_FILE}
-        INTERFACE_INCLUDE_DIRECTORIES ${ARROW_DOWNLOAD_PATH}/include
-    )
-
-    # Hack to propagate INTERFACE_INCLUDE_DIRECTORIES.
-    # via https://stackoverflow.com/a/47358004
-    file(MAKE_DIRECTORY ${ARROW_DOWNLOAD_PATH}/include)
+    include(download_and_build_arrow.cmake)
```

Comment:
Does this expose the variables necessary to implement https://github.com/kyle-figure/bazel-minimal-rerun/blob/main/third_party/rerun/patches/cmake.patch ?

Reply:
tbh I'm deeply confused (and troubled) about the visibility scope of cmake variables, so I don't know. Worst case this requires more patches then